Mathlib/Topology/Algebra/Algebra.lean

```diff
@@ -574,6 +574,53 @@ theorem Subalgebra.topologicalClosure_comap_homeomorph (s : Subalgebra R A) {B :
   rw [w]
   exact f'.preimage_closure _
 
+variable (R)
+
+open Subalgebra
+
+/-- The topological closure of the subalgebra generated by a single element. -/
+def Algebra.elemental (x : A) : Subalgebra R A :=
+  (Algebra.adjoin R ({x} : Set A)).topologicalClosure
+
+@[deprecated (since := "2024-11-05")] alias Algebra.elementalAlgebra := Algebra.elemental
+
+namespace Algebra.elemental
+
+@[aesop safe apply (rule_sets := [SetLike])]
+theorem self_mem (x : A) : x ∈ elemental R x :=
+  le_topologicalClosure _ <| self_mem_adjoin_singleton R x
+
+@[deprecated (since := "2024-11-05")] alias _root_.Algebra.self_mem_elementalAlgebra := self_mem
+
+theorem le_of_mem {x : A} {s : Subalgebra R A} (hx : x ∈ s)
+    (hs : IsClosed (s : Set A)) :
+    elemental R x ≤ s :=
+  topologicalClosure_minimal _ (adjoin_le <| by simpa using hx) hs
```

Comment:
Could you also add the `iff` statement ? These tend to be very useful when rewriting.